Scripts/extras/pull_and_build_from_git.py

```diff
@@ -52,6 +52,8 @@
                           to restrict building to a specific configuration rather than building all configurations 
                           (provided by the default value: ['Debug', 'Release'])
 * patch_file            : (optional) Option patch file to apply to the synced source before performing a build
+* local_repo            : (optional) Option to provide a path to the project source rather than getting it from github
+* git_skip              : (optional) Option to skip all git commands, requires local_repo
```

Comment:
sorry to ask for another change, but could you change this name to something like "source_directory" instead of "local_repo"?  And then remove git_skip because "source_directory" would imply it?

The reason for this is that a local repo is actually a git concept.
In fact, if you instead of doing this change this: instead

On line 337, in clone_to_local function, after the line 
```
relative_src_dir = self.src_folder.name
```
add:
```
relative_src_dir = self.src_folder.name

if not self.package_info.git_url.startswith("http") and not self.package_info.git_url.startswith("git"):
            self.package_info.git_url = str(pathlib.Path(self.package_info.git_url).absolute())
```

Then whats going to happen is git will happily take absolute or relative paths to folders containing a git repository and treat it as local automatically.

eg, if you have a local git repository in d:\myprojects\blah (you have called git init, git add . , git commit -m initial)

if you then do a 'git clone "d:/myprojects/blah" it will actualy automatically understand it doesn't have to go to the internet.  

Perhaps thats a better way of working then anyway since it also allows you do stil use this script to apply patches?

Reply:
I disagree that using local_repo or source_folder always means I want to skip the git commands.

As an example I will be working out the 3p instructions for FreeType soon, and I will use the pull_and_build script to sync down the repo, but I want to specify where to put it. In that case I was to specify source_folder but not skip the git commands.